### PR TITLE
Eliminate TODOs, bump version to 0.2.0, add tests

### PR DESCRIPTION
**Pull request by mgradwohl**
## Description

Comprehensive code cleanup that eliminates TODOs, bumps version to 0.2.0, and adds new test coverage.

**Cleanup:**
- Replaced 15 TODOs with GitHub issue #287 (Options dialog)
- Deleted temporary review documentation files (PR_REVIEW_COMMENTS.md, GITHUB_ISSUES_FROM_REVIEW.md, full_code_review.md, etc.)
- Version bump 0.1.0 → 0.2.0

**Code improvements:**
- Added GetSystemMetrics validation in Window.cpp (P2-14 fix)
- Changed search buffer from `std::array<char, 256>` to `std::string` for dynamic sizing (P2-9 fix)
- Applied clang-tidy fixes across Platform layer
- Added `[[nodiscard]]` attributes where appropriate
- Fixed hex string parsing in ThemeLoader
- Replaced refresh interval literals with SamplingConfig constants

**New tests:**
- `test_UserConfig.cpp`: 27 tests for UserSettings struct
- `test_Numeric.cpp`: Floating-point comparison edge cases

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Build/CI improvement

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines
- [x] My code follows the project's coding standards
- [x] I have run `pre-commit run --all-files` or installed pre-commit hooks
- [x] I have run `clang-format` on my changes
- [x] I have run `clang-tidy` and addressed any warnings
- [x] I have added tests that prove my fix/feature works
- [x] All new and existing tests pass
- [x] I have updated documentation as needed

## Testing

```bash
# Commands used to test
cmake --preset win-debug
cmake --build --preset win-debug
ctest --preset win-debug
```

**Results:** All 285 tests pass (1 intentionally skipped)

## Additional Notes

- Related issue: #287 (Options Dialog - created to track the eliminated TODOs)
- clang-format: 121 files formatted
- clang-tidy: Passed with no warnings
- Version shown in About box will now display 0.2.0

### CODE REVIEW

File: tests/App/test_UserConfig.cpp [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    UserSettings original;
+    original.themeId = "move-theme";
+    original.imguiLayout = "some large layout data that would benefit from move";
+
+    // Move
+    UserSettings moved = std::move(original);
+
+    EXPECT_EQ(moved.themeId, "move-theme");
+    EXPECT_FALSE(moved.imguiLayout.empty());
+}
```
[the remark this answers is not in the document]
Comment (mgradwohl) [reply tests/App/test_UserConfig.cpp]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/mgradwohl/tasksmack/pull/288#discussion_r2650037290)
